File: Cpp/1-50/12.cpp
#include <iostream>
#include <math.h>
using namespace std;
int main()
{
    long long int n, soniTub = 1;
    cin >> n;
    bool ha = true;
    for (int j = 3; j <= n; j += 2)
    {
        ha = true;
        for (int i = 2; i <= sqrt(j); i++)
        {
            if (j % i == 0)
            {
                ha = false;
                break;
            }
        }
        if (ha)
            soniTub++;
    }
    if (soniTub % 2 != 0)
        cout << "Ali";
    else
        cout << "Bobur";
}

File: C sharp/1-50/12.cs
using System;

class Program {
    static void Main(string[] args) {
        int n = int.Parse(Console.ReadLine());
        int k = 0;
        for (int i = 2; i <= n; i++) {
            bool x = true;
            for (int j = 2; j <= Math.Sqrt(i); j++) {
                if (i % j == 0) {
                    x = false;  
                    break;
                }
            }
            if (x) {
                k++;
            }
        }
        if (k % 2 == 0) {
            Console.WriteLine("Bobur");
        } else {
            Console.WriteLine("Ali");
        }
    }
}